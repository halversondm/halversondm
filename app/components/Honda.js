/**
 * Created by Daniel on 6/26/2016.
 */
"use strict";

import React from "react";
import PhotoGallery from "./PhotoGallery";

const Honda = React.createClass({

  render() {
    return <div>
      <h2 className="text-primary">1997 Honda Shadow 750 ACE</h2>
      <p>In August 2007, I bought this Honda Shadow. The bike came with the
        saddle bags, custom forward controls, and windshield. It was a pretty
        good find for the price. I didn't love the single-front disc and rear
        drum brake, but otherwise it was a solid bike. Unfortunately, I lost the
        bike to an accident on I294 on the way to Harley's 105th Anniversary
        Celebration in 2008.</p>
      <PhotoGallery perPage={6} totalPhotos={6} filePrefix="images/hondaPhotos/photo" fileSuffix=".jpg" />
    </div>;
  }
});

export default Honda;
