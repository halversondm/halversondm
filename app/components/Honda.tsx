/**
 * Created by Daniel on 6/26/2016.
 */
import * as React from "react";
import PhotoGallery from "./PhotoGallery";
import { type ReactNode } from "react";

export default function Honda(): ReactNode {
  return (
    <div>
      <h2 className="text-primary">1997 Honda Shadow 750 ACE</h2>
      <p>
        In August 2007, I bought this Honda Shadow. The bike came with the
        saddle bags, custom forward controls, and windshield. It was a pretty
        good find for the price. I didn&apos;t love the single-front disc and
        rear drum brake, but otherwise it was a solid bike. Unfortunately, I
        lost the bike to an accident on I294 on the way to Harley&apos;s 105th
        Anniversary Celebration in 2008.
      </p>
      <PhotoGallery
        perPage={6}
        totalPhotos={6}
        filePrefix="images/hondaPhotos/photo"
        fileSuffix=".jpg"
      />
    </div>
  );
}
