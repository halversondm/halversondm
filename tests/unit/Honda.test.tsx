import React from "react";
import { render } from '@testing-library/react';
import Honda from "../../app/components/Honda";

test("Honda component", () => {
    const unit = <Honda />;
    const component = render(unit);
    const tree = component.container;
    expect(tree.toString()).toMatchSnapshot();
});