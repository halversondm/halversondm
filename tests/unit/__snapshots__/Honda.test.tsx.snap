// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Honda component 1`] = `"<div><div><h2 class=\\"text-primary\\">1997 Honda Shadow 750 ACE</h2><p>In August 2007, I bought this Honda Shadow. The bike came with the saddle bags, custom forward controls, and windshield. It was a pretty good find for the price. I didn't love the single-front disc and rear drum brake, but otherwise it was a solid bike. Unfortunately, I lost the bike to an accident on I294 on the way to Harley's 105th Anniversary Celebration in 2008.</p><div id=\\"photoGallery\\"><ul class=\\"row photo-gallery\\"><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"0\\" class=\\"img-fluid\\" src=\\"images/hondaPhotos/photo1.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"1\\" class=\\"img-fluid\\" src=\\"images/hondaPhotos/photo2.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"2\\" class=\\"img-fluid\\" src=\\"images/hondaPhotos/photo3.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"3\\" class=\\"img-fluid\\" src=\\"images/hondaPhotos/photo4.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"4\\" class=\\"img-fluid\\" src=\\"images/hondaPhotos/photo5.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"5\\" class=\\"img-fluid\\" src=\\"images/hondaPhotos/photo6.jpg\\"></li></ul><div class=\\"row\\"><div><button type=\\"button\\" data-id=\\"1\\" class=\\"btn btn-primary\\">Page 1</button></div></div></div></div></div>"`;
